### ISSUE TITLE
[frontend] Filter pipeline text box shows error when typing anything in it

### ISSUE THREAD

**Issue by czchen** [kubeflow/pipelines#10241]
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)? kustomization
*  KFP version:  2.0.3

### Steps to reproduce

* Select `Runs` on left sidebar.
* Select `+ Create run` on top right sidebar.
* Select `Choose` in `Pipeline` input box.
* `Type a` in `Filter pipelines` input text.
* An `Error retrieving resources` is poped up, with the following error message

```
List request failed with:
{"error":"Failed to list pipelines due invalid list options: pageToken: , pageSize: 10, sortBy: created_at desc, filter: %7B%22predicates%22%3A%5B%7B%22key%22%3A%22name%22%2C%22op%22%3A%22IS_SUBSTRING%22%2C%22string_value%22%3A%22a%22%7D%5D%7D: Invalid input error: failed to parse valid filter from \"%7B%22predicates%22%3A%5B%7B%22key%22%3A%22name%22%2C%22op%22%3A%22IS_SUBSTRING%22%2C%22string_value%22%3A%22a%22%7D%5D%7D\": unknown field \"op\" in kubeflow.pipelines.backend.api.v2beta1.Predicate","code":3,"message":"Failed to list pipelines due invalid list options: pageToken: , pageSize: 10, sortBy: created_at desc, filter: %7B%22predicates%22%3A%5B%7B%22key%22%3A%22name%22%2C%22op%22%3A%22IS_SUBSTRING%22%2C%22string_value%22%3A%22a%22%7D%5D%7D: Invalid input error: failed to parse valid filter from \"%7B%22predicates%22%3A%5B%7B%22key%22%3A%22name%22%2C%22op%22%3A%22IS_SUBSTRING%22%2C%22string_value%22%3A%22a%22%7D%5D%7D\": unknown field \"op\" in kubeflow.pipelines.backend.api.v2beta1.Predicate","details":[{"@type":"type.googleapis.com/google.rpc.Status","code":3,"message":"failed to parse valid filter from \"%7B%22predicates%22%3A%5B%7B%22key%22%3A%22name%22%2C%22op%22%3A%22IS_SUBSTRING%22%2C%22string_value%22%3A%22a%22%7D%5D%7D\": unknown field \"op\" in kubeflow.pipelines.backend.api.v2beta1.Predicate"}]}
```

### Expected result

There shall be no error message when typing in  `Filter pipelines` input box.

### Materials and Reference

---

Impacted by this bug? Give it a 👍.

**Comment by lethee** [comment]
I have same issue.

Predicate:
```
{
    "predicates": [
        {
            "key": "name",
            "op": "IS_SUBSTRING",
            "string_value": "d"
        },
        {
            "key": "storage_state",
            "operation": "NOT_EQUALS",
            "string_value": "ARCHIVED"
        }
    ]
}
```

My env:
```
{
    "buildDate": "Fri Oct 27 04:09:58 UTC 2023",
    "frontendCommitHash": "58ce09e07d031964905020c749e77bf0f37e83d4",
    "frontendTagName": "2.0.3",
    "apiServerReady": true,
    "apiServerCommitHash": "58ce09e07d031964905020c749e77bf0f37e83d4",
    "apiServerTagName": "2.0.3",
    "apiServerMultiUser": true,
    "multi_user": true
}
```

<img width="747" alt="Screenshot 2024-01-04 at 11 31 26 AM" src="https://github.com/kubeflow/pipelines/assets/68858/66414f88-53af-409b-98d1-6ae698e99938">


I think problem is on https://github.com/kubeflow/pipelines/blob/master/frontend/src/components/CustomTable.tsx#L545

**Comment by tom-pavz** [comment]
We are still seeing this exact bug in KFP backend `2.1.0`

**Comment by thesuperzapper** [comment]
@droctothorpe given your work on https://github.com/kubeflow/pipelines/pull/10797, I wonder if you have also seen this issue?

It seems like the first time you try and create a new run by:

1. Going to the "Runs" page
2. Clicking "Create Run"
3. Clicking "Choose" pipeline
4. Finding a pipeline and clicking "Use this Pipeline"

You will get a crazy slow response, or sometimes you will get an error like what this issue raises.

**Comment by droctothorpe** [comment]
We've noticed this as well! Haven't had a chance to debug it yet though.

**Comment by thesuperzapper** [comment]
@droctothorpe The issue I am talking about in https://github.com/kubeflow/pipelines/issues/10241#issuecomment-2156129113 seems to be different from the one raised by @czchen in this issue, sorry. I have raised a separate issue for that: https://github.com/kubeflow/pipelines/issues/10897

The issue is raised by @czchen is still very important because it's actually a broken feature (filtering literally does not work in V2).

**Comment by boarder7395** [comment]
Tested this in 2.2.0 and search is still broken. I took a look at https://github.com/kubeflow/pipelines/pull/10358 and while if fixes the create run page, it breaks the clone from run page.

Create Run Page Error on 2.2.0:
![Screenshot 2024-08-15 at 2 39 08 PM](https://github.com/user-attachments/assets/1a1801de-0dfb-44e5-a408-fcf47cdd919e)

Clone Run Page Error with fix #10358 
![Screenshot 2024-08-15 at 3 11 05 PM](https://github.com/user-attachments/assets/9b0ad121-737e-4d8c-9c35-20941ddda3c7)

**Comment by tom-pavz** [comment]
@HumairAK  This still appears to be an issue when typing in the `Experiments` filter box:

<img width="2145" height="1126" alt="Image" src="https://github.com/user-attachments/assets/52abd565-b784-4bb1-89da-6ca7049c7288" />
